src/main/java/com/yandex/money/api/typeadapters/showcase/ComponentsTypeProvider.java

```diff
@@ -0,0 +1,63 @@
+package com.yandex.money.api.typeadapters.showcase;
+
+import com.google.gson.JsonElement;
+import com.yandex.money.api.model.showcase.components.Component;
+import com.yandex.money.api.model.showcase.components.container.Group;
+import com.yandex.money.api.model.showcase.components.container.Paragraph;
+import com.yandex.money.api.model.showcase.components.uicontrol.Amount;
+import com.yandex.money.api.model.showcase.components.uicontrol.Checkbox;
+import com.yandex.money.api.model.showcase.components.uicontrol.Date;
+import com.yandex.money.api.model.showcase.components.uicontrol.Email;
+import com.yandex.money.api.model.showcase.components.uicontrol.Month;
+import com.yandex.money.api.model.showcase.components.uicontrol.Number;
+import com.yandex.money.api.model.showcase.components.uicontrol.Select;
+import com.yandex.money.api.model.showcase.components.uicontrol.Submit;
+import com.yandex.money.api.model.showcase.components.uicontrol.Tel;
+import com.yandex.money.api.model.showcase.components.uicontrol.Text;
+import com.yandex.money.api.model.showcase.components.uicontrol.TextArea;
+import com.yandex.money.api.typeadapters.showcase.uicontrol.ComponentTypeAdapter;
+
+import java.lang.reflect.Type;
+import java.util.HashMap;
+
+/**
+ * @author Anton Ermak (ermak@yamoney.ru)
+ */
+public final class ComponentsTypeProvider {
+
+    private static final HashMap<Component.Type, Type> TYPE_MAPPING = new HashMap<>();
```

Comment:
There's a risk that this map can be changed by mistake. Please make it unmodifiable.